code/modules/mob/living/carbon/human/life.dm

```diff
@@ -485,6 +485,8 @@
 	if(on_fire)
 		var/thermal_protection = get_thermal_protection()
 
+		if(wear_suit && wear_suit.flags & AUTOEXTINGUISH)
+			wear_suit.auto_extinguish(src)

```

Comment:
This shouldn't be here; it should be on plasmamen's `Life` or where the code is that handles them igniting without their suit.